File: array/1313_decompress_run_length_encoded_list/1313_decompress_run_length_encoded_list.py
from typing import List

class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        ans = list()
        for i in range(0, len(nums), 2):
            ans.extend([nums[i + 1]] * nums[i])
        return ans


nums = [1,2,3,4]
res = Solution().decompressRLElist(nums)
print(res)

File: array/1313_decompress_run_length_encoded_list/1313_decompress_run_length_encoded_list.go
package main

import "fmt"

func decompressRLElist(nums []int) []int {
	var res []int
	l := len(nums)
	i := 0
	for i < l {
		for 0 < nums[i] {
			nums[i] -= 1
			res = append(res, nums[i+1])
		}
		i += 2
	}
	return res
}

func main() {
	nums := []int{1, 2, 3, 4}
	res := decompressRLElist(nums)
	fmt.Println(res)
}
